crates/bevy_ecs/src/storage/table.rs

```diff
@@ -669,7 +673,8 @@ impl TableBuilder {
 /// [`Component`]: crate::component::Component
 /// [`World`]: crate::world::World
 pub struct Table {
-    columns: ImmutableSparseSet<ComponentId, Column>,
+    columns: Vec<Column>,
```

Comment:
@james-j-obrien knows more here.

I think a good first solution to fix memory issues would indeed be to just swap the SparseSet for a Hashmap.
I think the idea was that HashMap have relatively poor performance for iteration

Reply:
> Btw @chescock have you joined the bevy discord? It would be great to chat more about a path to relations!

Yup!  I'm quartermeister there, and I've been lurking on #ecs-dev and the 🌈 working group room!

Reply:
Storing a HashMap in the table is almost equivalent to storing it in the component index, however in a post-relations future where tables are being created and deleted much more frequently it's economical to try and keep the setup and teardown costs low. It may also be convenient for later optimizations like moving the sparse sets into the component index, that way we only have one codepath for both storages and additionally we are not looking up the info and then the sparse set separately.

While benchmarks haven't been run yet there's a decent chance we want to steal some of flecs' optimizations here and keep a low id lookup array in both the table and the component index.